### DISCUSSION TITLE
Unexpected behavior Repeat with Aff using Point-Free Style

### DISCUSSION THREAD

**Discussion by seungyongshim** [louthy/language-ext#1228] [Q&A]
Excuse, me.

I don't understand. 

### Test

```csharp
f = Aff(() => AddAsync().ToValue()) // correctly
```
vs
```csharp
f = Aff(AddAsync().ToValue) // memoization ??
```

### Result
   * https://github.com/seungyongshim/poc-aff-retry/blob/master/src/ConsoleApp1/AffCase1.cs#L14
   * https://github.com/seungyongshim/poc-aff-retry/blob/master/src/ConsoleApp1/AffCase2.cs#L14
```
AffCase1: 2
AffCase2: 1
```

**Comment by louthy** [comment]
I'm away on vacation for a week, so can't look into this now, but the way you declare `f` above is not the same in your linked examples.  

Either define `f` as a free variable or don't, mixing them could well give different results.  

However, I'd need to check the code on my return.
